src/Analyzer.Utilities/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -93,8 +93,18 @@ public static IEnumerable<ITypeSymbol> GetBaseTypesAndThis(this ITypeSymbol type
             }
         }
 
-        public static bool DerivesFrom(this INamedTypeSymbol symbol, INamedTypeSymbol candidateBaseType)
+        public static bool DerivesFrom(this ITypeSymbol symbol, ITypeSymbol candidateBaseType, bool baseTypesOnly = false)
```

Comment:
Well, I obviously don't understand Roslyn well enough to comment on the design of those interfaces (although I tend to believe there's a reason behind this decision, which is usually the case). For that matter, I'd suggest you to contact compiler team directly (or file an issue in Roslyn repo.)

That being said, I merely tried to point out to you that based on current Roslyn implementation (that is, 1.2 and older), this change will **only** cause overhead. So please fix this. Thanks.